frontend/simple/views/containers/Proposals.vue

```diff
@@ -55,138 +32,100 @@
           own.push(proposal)
-        } else if (proposal.data.userVote === null) {
-          notVoted.push(proposal)
-        } else if (proposal.data.userVote !== null) {
+        } else if (proposal.myVote) {
           alreadyVoted.push(proposal)
+        } else {
+          notVoted.push(proposal)
         }
       }
-
       return { notVoted, alreadyVoted, own }
-    },
-    allVoted () {
-      const { notVoted, alreadyVoted } = this.groupProposals
-
-      return alreadyVoted.length && notVoted.length === 0
     }
   },
   methods: {
-    handleVoteAgainst () {
-      console.log('TODO Logic - The user voted against')
+    async handleVote (hash, vote) {
+      try {
+        // Create a vote against the proposal
+        const groupId = this.$store.state.currentGroupId
+        const voteAction = await sbp('gi/contract/create-action', 'GroupProposalVote', {
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash,
+          vote
+        }, groupId)
+        await sbp('backend/publishLogEntry', voteAction)
+      } catch (ex) {
+        // TODO: save to error log
+        console.error(ex)
+        throw new Error()
+      }
+    },
+    handleVoteAgainst (hash) {
+      this.handleVote(hash, -1)
     },
-    handleVoteFor () {
-      console.log('TODO Logic - The user voted for')
+    async handleVoteFor (hash) {
+      this.handleVote(hash, 1)
+      // If the vote passes fulfill the action
+      // TODO better place for this?
+      const proposal = this.$store.getters.proposalData(hash)
+      const memberCount = Object.entries(this.currentGroupState.profiles).length
+      const threshold = Math.ceil(proposal.voteRule.threshold * memberCount)
+      if (proposal.votes.filter(vote => vote.vote === 1).length + 1 >= threshold) {
+        await this.handleVotePassed(proposal)
+      }
     },
-    handleCloseProposal () {
-      console.log('TODO Logic - The proposal was closed')
+    async handleVotePassed (proposal) {
+      try {
+        // If the vote passes fulfill the action
+        // TODO: this is poorly implementated. do not create poposals in this manner.
+        for (let step of proposal.actions) {
+          let actData = JSON.parse(step.action)
+          let entry = await sbp('gi/contract/create-action', step.type, actData, step.contractID)
+          await sbp('backend/publishLogEntry', entry)
+        }
+      } catch (ex) {
+        // TODO: save to error error
+        console.error(ex)
+        throw new Error()
+      }
+    },
+    async handleCloseProposal (hash) {
+      try {
+        // Create proposal close event
+        const groupId = this.$store.state.currentGroupId
+        const close = await sbp('gi/contract/create-action', 'GroupCloseProposal', {
+          username: this.currentUserIdentityContract.attributes.name,
+          proposalHash: hash
+        }, groupId)
+        await sbp('backend/publishLogEntry', close)
```

Comment:
Similarly, this logic can be moved into the `proposals.js` file as well, which will contain a large map keyed by `proposalType`, with values for all the UI messages, and functions to do stuff like the above. It will greatly reduce the size of this file and create a single place to visit for proposal logic.

I am also thinking that having the "last user who votes send the action steps" is the wrong approach. Instead they should just cast their vote, and then the logic in `group.js` and `proposals.js` should automatically handle the changes once the voting rule declares the vote as passed. Let's discuss this.